processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -23,7 +37,101 @@ func (dp Histogram) Add(in Histogram) Histogram {
 	panic("todo")
 }
 
-// nolint
 func (dp ExpHistogram) Add(in ExpHistogram) ExpHistogram {
-	panic("todo")
+	switch {
+	case dp.Timestamp() >= in.Timestamp():
+		fail("out of order")
+	case dp.Scale() != in.Scale():
+		fail("scale changed")
+	case dp.ZeroCount() != in.ZeroCount():
+		fail("zero count changed")
+	}
+
+	aggregate := func(dpBuckets, inBuckets pmetric.ExponentialHistogramDataPointBuckets) {
+		var (
+			dp   = Buckets{data: dpBuckets.BucketCounts(), offset: int(dpBuckets.Offset())}
+			in   = Buckets{data: inBuckets.BucketCounts(), offset: int(inBuckets.Offset())}
+			aggr = Buckets{data: pcommon.NewUInt64Slice()}
+		)
+		aggr.offset = int(min(dpBuckets.Offset(), inBuckets.Offset()))
+		if aggr.offset == dp.offset {
+			aggr.data = dp.data
+		}
+		aggr.EnsureLen(max(dp.Len(), in.Len()))
+
+		for i := 0; i < aggr.Len(); i++ {
+			aggr.SetAt(i, dp.At(i)+in.At(i))
+		}
+
+		aggr.CopyTo(dpBuckets)
+	}
+
+	aggregate(dp.Positive(), in.Positive())
+	aggregate(dp.Negative(), in.Negative())
+
+	count, sum := dp.stats()
+	dp.SetCount(count)
+	dp.SetSum(sum)
+	dp.SetTimestamp(in.Timestamp())
+	if dp.HasMin() {
+		dp.SetMin(min(dp.Min(), in.Min()))
+	}
+	if dp.HasMax() {
+		dp.SetMax(max(dp.Max(), in.Max()))
+	}
+
+	return dp
+}
+
+func (dp ExpHistogram) stats() (count uint64, sum float64) {
```

Comment:
```
	// count is the number of values in the population. Must be
	// non-negative. This value must be equal to the sum of the "bucket_counts"
	// values in the positive and negative Buckets plus the "zero_count" field.
	Count uint64 `protobuf:"fixed64,4,opt,name=count,proto3" json:"count,omitempty"`
```

What is called `sum` in this function must be `count` instead.
And what is called `count` is counting non-empty buckets, which is not really part of a data model.
And I don't think you can calculate `sum` from the datapoint's buckets because the original values are already lost.

Reply:
Also the negative buckets and the zero count are ignored.

Reply:
ahh, looks like I misread the spec here. good catch!